### PR TITLE
Check module name conflict when ignoring casing

### PR DESCRIPTION
**Pull request by meteorcloudy**
Addressing https://github.com/bazelbuild/bazel-central-registry/pull/1591#discussion_r1517371565

### CODE REVIEW

**Comment by meteorcloudy** [comment]
Tested on Linux:
```
+++ Validating zlib@1.2.11


BcrValidationResult.FAILED: Module name conflict found: zlib and 1 other modules have the same name when ignoring case sensitivity.

BcrValidationResult.FAILED: Module name conflict found: Zlib and 1 other modules have the same name when ignoring case sensitivity
```

File: tools/bcr_validation.py [hunk 1]
```diff
@@ -265,8 +265,22 @@ def validate_presubmit_yml(self, module_name, version):
     if report_num_new == report_num_old:
       self.report(BcrValidationResult.GOOD, "The presubmit.yml file is valid.")
 
+  def verify_module_name_conflict(self):
+    """Verify no module name conflict when ignoring case sensitivity."""
+    module_names = self.registry.get_all_modules()
+    lower_module_names = [name.lower() for name in module_names]
+    conflict_found = False
+    for name in module_names:
+      count = lower_module_names.count(name.lower())
```
Comment (fmeum) [inline tools/bcr_validation.py:274]: This is quadratic in the number of modules, which may become an issue in the future. How about grouping by normalized module name in a dict instead?

Comment (meteorcloudy) [reply tools/bcr_validation.py]: Good point, done!

```
BcrValidationResult.FAILED: Module name conflict found: zlib, Zlib
```

File: tools/bcr_validation.py [hunk 2]
```diff
@@ -265,8 +265,29 @@ def validate_presubmit_yml(self, module_name, version):
...
+    """Verify no module name conflict when ignoring case sensitivity."""
+    module_names = self.registry.get_all_modules()
+    conflict_found = False
+    module_group = {}
+    for name in module_names:
+      lower_name = name.lower()
+      if lower_name in module_group:
+          module_group[lower_name].append(name)
+      else:
+          module_group[lower_name] = [name]
```
Comment (fmeum) [inline tools/bcr_validation.py:278]:
```suggestion
      module_group.setdefault(name.lower(), []).append(name)
```

Comment (fmeum) [reply tools/bcr_validation.py:278]: If fine either way though.

**Review by fmeum — APPROVED** [review]

File: tools/bcr_validation.py (hunk 2 above)
Comment (meteorcloudy) [reply tools/bcr_validation.py]: I do prefer cleaner code, thanks!

**Pull request merged**